ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-popover.js

```diff
@@ -0,0 +1,243 @@
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import FormField from '../../ui/form-field';
+import Box from '../../ui/box';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import I18nValue from '../../ui/i18n-value';
+import {
+  decGWEIToHexWEI,
+  hexWEIToDecGWEI,
+} from '../../../helpers/utils/conversions.util';
+import {
+  divideCurrencies,
+  multiplyCurrencies,
+} from '../../../../shared/modules/conversion.utils';
+import { TYPOGRAPHY } from '../../../helpers/constants/design-system';
+import { SECONDARY } from '../../../helpers/constants/common';
+import { useGasFeeContext } from '../../../contexts/gasFee';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { useUserPreferencedCurrency } from '../../../hooks/useUserPreferencedCurrency';
+import { useCurrencyDisplay } from '../../../hooks/useCurrencyDisplay';
+import AdvancedGasFeeInputSubtext from './advanced-gas-fee-input-subtext';
+
+const AdvancedGasFeePopover = ({ onClose }) => {
+  const t = useI18nContext();
+  const {
+    maxPriorityFeePerGas,
+    maxFeePerGas,
+    estimatedBaseFee,
+    onManualChange,
+  } = useGasFeeContext();
+
+  const [editingInGwei, setEditingInGwei] = useState(false);
+  const [priorityFee, setPriorityFee] = useState(maxPriorityFeePerGas);
+
+  const estimatedBaseFeeInDecGWEI = hexWEIToDecGWEI(estimatedBaseFee, {
+    numberOfDecimals: 6,
+  });
+  const baseFeeMultiplier = divideCurrencies(
+    decGWEIToHexWEI(maxFeePerGas),
+    estimatedBaseFee,
+    {
+      numberOfDecimals: 6,
+      dividendBase: 16,
+      divisorBase: 16,
+      toNumericBase: 'dec',
+    },
+  );
+  const [maxBaseFeeMultiplier, setMaxBaseFeeMultiplier] = useState(
+    baseFeeMultiplier,
+  );
+
+  const baseFee = multiplyCurrencies(estimatedBaseFee, maxBaseFeeMultiplier, {
+    numberOfDecimals: 6,
+    multiplicandBase: 16,
+    multiplierBase: 10,
+    toNumericBase: 'dec',
+    fromDenomination: 'WEI',
+    toDenomination: 'GWEI',
+  });
+  const [maxBaseFeeGWEI, setMaxBaseFeeGWEI] = useState(baseFee);
+  const { currency, numberOfDecimals } = useUserPreferencedCurrency(SECONDARY);
+  const [, { value: baseFeeInFiat }] = useCurrencyDisplay(
+    decGWEIToHexWEI(baseFee),
+    { currency, numberOfDecimals },
+  );
+  const [, { value: priorityFeeInFiat }] = useCurrencyDisplay(
+    decGWEIToHexWEI(priorityFee),
+    { currency, numberOfDecimals },
+  );
+  const baseFeeFiatMessage =
+    parseFloat(baseFeeInFiat.split('$')[1]) >= 0.01
```

Comment:
Instead of taking the base fee (as a decimal), converting it to wei (as hex), converting it to a currency string, and then parsing that currency string, can we use `baseFee` directly? We would ideally want it to be a BigNumber so that we can just say `baseFee.gt(0.01)` here.